src/liballoc/boxed.rs

```diff
@@ -239,6 +239,16 @@ impl<T> Box<T> {
     pub fn pin(x: T) -> Pin<Box<T>> {
         (box x).into()
     }
+
+    /// Converts a `Box<T>` into a `Box<[T]>`
+    ///
+    /// This conversion does not allocate on the heap and happens in place.
+    ///
+    #[unstable(feature = "box_into_boxed_slice", issue = "71582")]
+    pub fn into_boxed_slice(boxed: Box<T>) -> Box<[T]> {
+        // *mut T and *mut [T; 1] have the same size and alignment
```

Comment:
This comment doesn't match the code. `*mut T` ≠ `*mut [T]`

Reply:
That comment is justifying the cast to an array of length 1. The slice comes after.

Reply:
What justifies the subsequent cast?

Reply:
That's just unsizing, but we could leave that off and let the `Box<[T; 1]>` implicitly coerce unsized in the return.